### DISCUSSION TITLE
Using the queryclient directly for a graphql mutation

### DISCUSSION THREAD

**Discussion by JessedeJonge** [TanStack/query#3186] [Q&A]
Im in a situation where I want to run a mutation outside a function component. For general use, I use hooks that are generated by codegen, but for this call I can not use hooks. I tried writing a function that just does a simple fetch with a post body, but I need an ENUM as a parameter for the mutation, and fetch seems to parse it as a string. Tried the following option:

```
export const initCall = async (info: DeviceInput) => {
  console.log(Device_Type.Android);
  const token = await getIdToken();
  const res = await fetch(`${API_URL}/query`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`,
    },
    body: JSON.stringify({
      query: `
      mutation {
        createDevice(data: {fcm_token: "${info.aspn_token}", device_type: ${Device_Type.Android}}) {
          id
        }
      }`,
    }),
  })
    .then((res) => res.json())
    .then((res) => console.log(res.data));
};

```

Device_type android is an enum generated by graphql codegen. I have also tried a string.

so, I was thinking; maybe the queryclient can handle this better? If so, how can i use it for post calls?

**Comment by TkDodo** [comment]
I think you can execute a mutation manually, but it won't solve your problem because you would still need provide a `mutationFn`, which actually fires off the network request (via fetch, axios, graphql-request or something like that).
